linera-base/src/identifiers.rs

```diff
@@ -142,20 +151,58 @@ impl ChainDescription {
 #[cfg_attr(with_testing, derive(Default))]
 pub struct ChainId(pub CryptoHash);
 
-/// A blob ID.
+/// A content addressed blob ID i.e. the hash of the Blob.
 #[derive(Eq, PartialEq, Ord, PartialOrd, Clone, Copy, Hash, Debug, Serialize, Deserialize)]
 #[cfg_attr(with_testing, derive(Default))]
 pub struct BlobId(pub CryptoHash);
 
+impl BlobId {
+    /// Creates a new `BlobId` from a `Blob`
+    pub fn new(blob: &Blob) -> Self {
+        BlobId(CryptoHash::new(blob))
+    }
+}
+
+#[cfg(with_testing)]
+impl Arbitrary for BlobId {
+    type Parameters = ();
+    type Strategy = strategy::Map<VecStrategy<RangeInclusive<u8>>, fn(Vec<u8>) -> BlobId>;
+
+    fn arbitrary_with((): Self::Parameters) -> Self::Strategy {
+        use generic_array::typenum::Unsigned;
+        type HasherOutputSize = <sha3::Sha3_256 as sha3::digest::OutputSizeUser>::OutputSize;
+
+        vec(u8::MIN..=u8::MAX, HasherOutputSize::to_usize()).prop_map(|vector| {
+            BlobId(CryptoHash::new_from_bytes(
+                generic_array::GenericArray::clone_from_slice(&vector[..]),
+            ))
+        })
+    }
+}
+
 /// A blob.
-#[derive(Serialize, Deserialize)]
+#[derive(Eq, PartialEq, Debug, Hash, Clone, Serialize, Deserialize)]
 pub struct Blob {
     #[serde(with = "serde_bytes")]
     bytes: Vec<u8>,
 }
 
 impl BcsSignable for Blob {}
 
+impl Blob {
+    /// Loads blob from a file.
+    pub async fn load_from_file(path: impl AsRef<std::path::Path>) -> std::io::Result<Self> {
+        let bytes = std::fs::read(path)?;
+        Ok(Blob { bytes })
```

Comment:
The `BlobId` would be computed here.

Reply:
Do you mean we should compute it here and memoize it?